test/json_unittest.cc

```diff
@@ -112,6 +130,8 @@ TEST(TrivialToStringTest, StringOneChar) {
 TEST(TrivialParseTest, ArrayEmpty) {
   GuardJsonException([](){
     json::value v = json::Parser::FromString("[]");
+    EXPECT_TRUE(v->Is<json::Array>());
+    EXPECT_EQ(json::ArrayType, v->type());
     EXPECT_TRUE(v->As<json::Array>()->empty());
   });
```

Comment:
For the nonempty case, can you also check that you can extract a `json::value` using `operator[]` (i.e., `(*array)[0]`) and `at` (i.e., `array->at(0)`)?